aspnetcore/security/data-protection/compatibility/cookie-sharing.md

```diff
@@ -1,112 +1,184 @@
 
-## Sharing authentication cookies between applications
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/security/data-protection/compatibility/cookie-sharing/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
 
-To share authentication cookies between two different ASP.NET Core applications, configure each application that should share cookies as follows.
+The sample illustrates cookie sharing across three apps that use cookie authentication:
 
-In your configure method, use the CookieAuthenticationOptions to set up the data protection service for cookies and the AuthenticationScheme to match ASP.NET 4.x.
+* ASP.NET Core 2.0 Razor Pages app without using [ASP.NET Core Identity](xref:security/authentication/identity)
+* ASP.NET Core 2.0 MVC app with ASP.NET Core Identity
+* ASP.NET Framework 4.6.1 MVC app with ASP.NET Identity
 
-If you're using identity:
+In the examples that follow:
+
+* The authentication cookie name is set to a common value of `.AspNet.SharedCookie`.
+* The `AuthenticationType` is set to `Identity.Application` either explicitly or by default.
+* [CookieAuthenticationDefaults.AuthenticationScheme](/dotnet/api/microsoft.aspnetcore.authentication.cookies.cookieauthenticationdefaults.authenticationscheme) is used as the authentication scheme. The constant resolves to a value of `Cookies`.
+* The cookie authentication middleware uses an implementation of [DataProtectionProvider](/dotnet/api/microsoft.aspnetcore.dataprotection.dataprotectionprovider). `DataProtectionProvider` provides data protection services for the encryption and decryption of authentication cookie payload data. The `DataProtectionProvider` instance is isolated from the data protection system used by other parts of the app.
+  * A common [data protection key](xref:security/data-protection/implementation/key-management) storage location is used. The sample app uses a folder named *KeyRing* at the root of the solution to hold the data protection keys.
+  * [DataProtectionProvider.Create(System.IO.DirectoryInfo)](/dotnet/api/microsoft.aspnetcore.dataprotection.dataprotectionprovider.create?view=aspnetcore-2.0#Microsoft_AspNetCore_DataProtection_DataProtectionProvider_Create_System_IO_DirectoryInfo_) accepts a [DirectoryInfo](/dotnet/api/system.io.directoryinfo) for use with authentication cookies. The sample app provides the path of the *KeyRing* folder to `DirectoryInfo`.
+  * [DataProtectionProvider](/dotnet/api/microsoft.aspnetcore.dataprotection.dataprotectionprovider) requires the [Microsoft.AspNetCore.DataProtection.Extensions](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection.Extensions/) NuGet package. The package is available to ASP.NET Core 2.0 apps when referencing the [Microsoft.AspNetCore.All](xref:fundamentals/metapackage) metapackage ([NuGet.org](https://www.nuget.org/packages/Microsoft.AspNetCore.All/)).
```

Comment:
Make it clear that the metapackage is intended for apps targeting .NET Core 2.0. Also call out explicitly that apps targeting .NET Framework must add this individual package.